lib/src/facts/aix/filesystem_resolver.cc

```diff
@@ -0,0 +1,116 @@
+#include <internal/facts/aix/filesystem_resolver.hpp>
+#include <internal/util/aix/odm.hpp>
+#include <internal/util/aix/vmount.hpp>
+#include <leatherman/file_util/file.hpp>
+#include <leatherman/logging/logging.hpp>
+
+#include <boost/algorithm/string.hpp>
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include <lvm.h>
+#include <odmi.h>
+#include <sys/cfgodm.h>
+#include <sys/statfs.h>
+
+using namespace std;
+using namespace facter::util::aix;
+namespace fu = leatherman::file_util;
+
+namespace facter { namespace facts { namespace aix {
+    filesystem_resolver::data filesystem_resolver::collect_data(collection& facts)
+    {
+        data result;
+        collect_filesystem_data(result);
+        collect_mountpoint_data(result);
+        collect_partition_data(result);
+        return result;
+    }
+
+    void filesystem_resolver::collect_filesystem_data(data& result)
+    {
+        fu::each_line("/etc/vfs", [&](const std::string& line) {
+            auto first_char = line.find_first_not_of(" \t");
+            if (line[first_char] != '#' &&  // skip comment lines
```

Comment:
I'm surprised that didn't blow up with my old version that just checked index `0`. Weird...